src/aioquic/quic/recovery.py

```diff
@@ -34,6 +34,69 @@ def __init__(self) -> None:
         self.sent_packets: Dict[int, QuicSentPacket] = {}
 
 
+class QuicCongestionControl:
+    """
+    New Reno congestion control.
+    """
+
+    def __init__(self) -> None:
+        self.bytes_in_flight = 0
+        self.congestion_window = K_INITIAL_WINDOW
+        self._congestion_recovery_start_time = 0.0
+        self._congestion_stash = 0
+        self._rtt_monitor = QuicRttMonitor()
+        self.ssthresh: Optional[int] = None
+
+    def on_packet_acked(self, packet: QuicSentPacket) -> None:
+        self.bytes_in_flight -= packet.sent_bytes
+
+        # don't increase window in congestion recovery
+        if packet.sent_time <= self._congestion_recovery_start_time:
+            return
+
+        if self.ssthresh is None or self.congestion_window < self.ssthresh:
+            # slow start
+            self.congestion_window += packet.sent_bytes
+        else:
+            # congestion avoidance
+            self._congestion_stash += packet.sent_bytes
+            count = self._congestion_stash // self.congestion_window
+            if count:
+                self._congestion_stash -= count * self.congestion_window
+                self.congestion_window += count * K_MAX_DATAGRAM_SIZE
+
+    def on_packet_sent(self, packet: QuicSentPacket) -> None:
+        self.bytes_in_flight += packet.sent_bytes
+
+    def on_packets_expired(self, packets: Iterable[QuicSentPacket]) -> None:
+        for packet in packets:
+            self.bytes_in_flight -= packet.sent_bytes
+
+    def on_packets_lost(self, packets: Iterable[QuicSentPacket], now: float) -> None:
+        lost_largest_time = 0.0
+        for packet in packets:
+            self.bytes_in_flight -= packet.sent_bytes
+            lost_largest_time = packet.sent_time
```

Comment:
Is `last_largest_time` supposed to be the maximum value of loss?

Reply:
`lost_largest_time` refers to the most recent `sent_time` of the lost packets. I should probably make a note to the effect that `packets` are sorted by sent time. Does that make sense?

Reply:
Yeah! Totally misread, disregard me :)